src/common/http/HttpClient.h

```diff
@@ -1,47 +1,63 @@
-/* Copyright (c) 2019 vesoft inc. All rights reserved.
+/* Copyright (c) 2022 vesoft inc. All rights reserved.
  *
  * This source code is licensed under Apache 2.0 License.
  */
 
-#ifndef COMMON_HTTPCLIENT_H
-#define COMMON_HTTPCLIENT_H
+#ifndef COMMON_HTTP_HTTPCLIENT_H_
+#define COMMON_HTTP_HTTPCLIENT_H_
+#include <memory>
+#include <mutex>
+#include <string>
 
-#include "common/base/Base.h"
 #include "common/base/StatusOr.h"
+#include "curl/curl.h"
 
 namespace nebula {
-namespace http {
+
+class CurlHandle {
+ public:
+  static CurlHandle* instance();
+
+ private:
+  CurlHandle();
+  ~CurlHandle();
+};
+
+struct HttpResponse {
+  CURLcode curlCode;
```

Comment:
`CURLcode` is a int not a enum? Didn't find it in this PR

Reply:
It's an integer and defined by libcurl